gridstatus/isone_api/isone_api.py

```diff
@@ -0,0 +1,439 @@
+import os
+import time
+
+import pandas as pd
+import requests
+
+from gridstatus import utils
+from gridstatus.base import NoDataFoundException
+from gridstatus.decorators import support_date_range
+from gridstatus.gs_logging import logger as log
+
+# API base URL
+BASE_URL = "https://webservices.iso-ne.com/api/v1.1"
+
+# Default page size for API requests
+DEFAULT_PAGE_SIZE = 1000
+
+ZONE_LOCATIONID_MAP = {
+    "NEPOOL": 32,
+    "INTERNALHUB": 4000,
+    "MAINE": 4001,
+    "NEWHAMPSHIRE": 4002,
+    "VERMONT": 4003,
+    "CONNECTICUT": 4004,
+    "RHODEISLAND": 4005,
+    "SEMASS": 4006,
+    "WCMASS": 4007,
+    "NEMASSBOST": 4008,
+}
```

Comment:
These were taken from the original POC code. We could have them match the API response verbatim (e.g. `.Z.MAINE`), leave them as is in this current cleaned up state, or some third option. The API really only cares about the `int` `Id` though.

Reply:
done